app/components/mfa_selection_component.html.erb

```diff
@@ -0,0 +1,24 @@
+<lg-validated-field>
```

Comment:
The custom element isn't really designed to be used independent of the ViewComponent implementation. An effect of this is there's no guaranteed it'd actually be defined in the first place. It might be currently working incidentally based on how Webpack is splitting JavaScript bundles, but this could break in the future.

Based on how this is implemented, I wonder if we could take advantage of the fact that ValidatedField is basically a wrapper for a [SimpleForm](https://github.com/heartcombo/simple_form) input field, and instead implement this as a [custom input](https://github.com/heartcombo/simple_form#custom-inputs), or a wrapper configuration [like we do with other checkboxes and radios](https://github.com/18F/identity-idp/blob/84f0706fb273a77fb140af733d8a43b34074fb0a/config/initializers/simple_form.rb#L34-L61).

This would also help avoid reimplementing error alert messages, which `ValidatedField` is already meant to be handling to create accessible linking between the field and the error message (not present here).

Reply:
Thanks for the feedback, I think I understood your comment, but let me know if I'm missing something.

I switched it to a div with a class for the scss to hit it. 

I initially tried to use ValidatedField for this but ran into a couple issues. 
1. The phone input is only invalid if no other input is selected, I wasn't able to figure out how to have validated field check other inputs to determine validation.
2. ValidatedField requires a :name to wire everything up properly. That name must be an attribute on the model, in this case the two_factor_options_form. However the form has a selection attribute that is an array, and name is only compatible with strings. I spent about a day trying to figure out how to convert from two_factor_options_form[:selection][] (or two_factor_options_form[:selection][:phone], etc.) to a usable :name.
3. Refactoring  two_factor_options_form to have the appropriate attributes so that I can use validated field adds a bunch of extra work for a requirement with a lot of urgency behind it and doesn't clear up the first issue for me.